### PR TITLE
trilium-desktop: add darwin support

### PR DESCRIPTION
**Pull request by eliandoran**
###### Description of changes

* Added `darwin` support for `trilium-desktop` by downloading the .zip archive from GitHub.
* Split `trilium-server` and `trilium-desktop` in two different files in order to avoid confusion when adding the `darwin` support (which is only for `trilium-desktop`).
* Changed the update script to match the new structure.
* Added myself as maintainer (see discussion below).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by eliandoran** [comment]
@FliegendeWurst , would you kindly have a look at this PR? I've left in draft because I haven't tested yet if I broke the Linux build, but I wanted to check with you whether you're OK with the refactoring.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]
Overall, the changes look good to me. I don't have a Mac to test the new package. I have a few suggestions:
- you can add yourself as a maintainer (given that you did a few updates and are adding darwing support now)
- `meta` information is now duplicated, it would be nice to combine these somehow

**Comment by eliandoran** [comment]
> Overall, the changes look good to me. I don't have a Mac to test the new package. I have a few suggestions:
> 
>     * you can add yourself as a maintainer (given that you did a few updates and are adding darwing support now)
> 
>     * `meta` information is now duplicated, it would be nice to combine these somehow

@FliegendeWurst , sounds good. I've done the requested changes and added myself as maintainer.

**Comment by eliandoran** [comment]
@FliegendeWurst , I've solved the merge conflicts. Could you please review?

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -16420,4 +16420,10 @@
     github = "srounce";
     githubId = 60792;
   };
+  eliandoran = {
+    email = "contact@eliandoran.me";
+    name = "Elian Doran";
+    github = "eliandoran";
+    githubId = 21236836;
+  };
```
Comment (FliegendeWurst) [inline maintainers/maintainer-list.nix:16428]: I think the entries here should be alphabetically sorted.

Comment (eliandoran) [reply maintainers/maintainer-list.nix]: @FliegendeWurst , good point. Fixed!

**Review by FliegendeWurst — APPROVED** [review]

**Comment by wegank** [comment]
@ofborg build trilium-desktop trilium-desktop.passthru.tests

**Pull request merged**